### ISSUE TITLE
Improve clarity of `possibly-missing-submodule` diagnostics

### ISSUE THREAD

**Issue by ZelphirKaltstahl** [astral-sh/ty#3540]
### Summary

I am importing `tkinter` as follows:

```python
import tkinter as tk
```

Then I use it for example as follows:

```python
        tk.messagebox.showinfo(
            title="No words",
            message="All good! None of the selected words are due for training.",
        )
```

`ty` complains about:

```
warning[possibly-missing-submodule]: Submodule `messagebox` might not have been i
mported
   --> src/lib/actions.py:169:9
    |
169 |         tk.messagebox.showinfo(
    |         ^^^^^^^^^^^^^
    |
help: Consider explicitly importing `tkinter.messagebox`
```

It would be annoying to have to import every little thing I need from `tkinter` one by one. I also don't see how explicitly importing everything adds any semantic information for the availability of some submodule. `ty` should go and look at the `tkinter` module and see by itself, that there is a thing called `messagebox` and that that thing has a thing `showinfo`. I don't see why it is incapable of figuring that out.

I also don't want to litter my code with tons of `# ty: ingore ...` inline comments. Currently I am just testing `ty` as a language server, to see whether it can replace another one for me.

When I use:

```python
from tkinter import (
    messagebox,
)
```

and subsequently use:

```python
        messagebox.showinfo(
            title="No words",
            message="All good! None of the selected words are due for training.",
        )
```

Then there are no complaints. I find this rather silly, because no typing information has been added that wasn't there before. All that has changed is, that now I am not using an alias for the import. But `ty` should be able to handle simple `as` aliases for imports. The only thing that happens is, that `ty` would force me to use its preferred import statements. Does it not deal well with `import ... as ...` imports in general?

### Version

`0.0.39`

**Comment by AlexWaygood** [comment]
Hey @ZelphirKaltstahl, sorry for the confusing diagnostic!

What ty is warning you about here is this issue: submodules on a package aren't always automatically loaded at runtime unless you explicitly import the submodule:

```pycon
Python 3.14.5 (main, May 10 2026, 19:20:57) [Clang 22.1.3 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import tkinter as tk
>>> tk.messagebox
Traceback (most recent call last):
  File "<python-input-5>", line 1, in <module>
    tk.messagebox
AttributeError: module 'tkinter' has no attribute 'messagebox'
>>> import tkinter.messagebox
>>> tk.messagebox
<module 'tkinter.messagebox' from '/Users/alexwaygood/.local/share/uv/python/cpython-3.14.5-macos-aarch64-none/lib/python3.14/tkinter/messagebox.py'>
```

So when ty is saying "Consider explicitly importing `tk.messagebox`, it's not saying "Add `from tkinter import messagebox` to the top of your code", it's saying "Add `import tkinter.messagebox` to the top of your code".

If your code works without this explicit import, it's likely because you're importing some other module that happens to import `tkinter.messagebox` transitively. At runtime, this causes the `messagebox` attribute to be cached on the `tkinter` package as a side effect, which then means that you can sometimes "get away" with not explicitly importing the submodule elsewhere. But this is very hard for type checkers to detect, and it's usually seen as bad style -- it's better to explicitly import the submodule.

I think we should definitely try to further improve our diagnostics here so it's easier for users to understand the recommendation we're making.

**Comment by ZelphirKaltstahl** [comment]
Oh OK, I see. I didn't know Python could behave that weirdly. Then I am going to adjust my imports, thank you!

**Comment by ZelphirKaltstahl** [comment]
Closing this issue now, as I think there is nothing more to add.
